server/utils/calendarDay.ts

```diff
@@ -0,0 +1,38 @@
+export default class CalendarDay {
+  private constructor(readonly day: number, readonly month: number, readonly year: number) {}
+
+  static fromComponents(day: number, month: number, year: number): CalendarDay | null {
+    return isValidDate(day, month, year) ? new CalendarDay(day, month, year) : null
+  }
+
+  static parseIso8601(val: string): CalendarDay | null {
+    const result = /^(\d{4})-(\d{2})-(\d{2})$/.exec(val)
+
+    if (result === null) {
+      return null
+    }
+
+    const year = Number(result[1])
```

Comment:
cosmetic - could be condensed to ```const [year, month, day] = result.map(Number)```? (i think)

Reply:
👍 (needs a `slice` too)